### PR TITLE
Unify Keras's cost function with the Estimator's

### PR DESCRIPTION
**Pull request by seemuch**
This PR does not complete the task. 

TODO: In the Estimator, the loss function also include an L2 cost, which
is currently missing from Keras's.

See: https://github.com/tfboyd/models/blob/resnet_perf_tweaks/official/resnet/resnet_run_loop.py#L316

### CODE REVIEW

File: official/resnet/keras/keras_imagenet_main.py [hunk 1]
```diff
@@ -154,7 +153,7 @@ def run_imagenet_with_keras(flags_obj):
   # in the graph.
   tf.keras.backend.set_learning_phase(True)
 
-  model.compile(loss='categorical_crossentropy',
+  model.compile(loss=softmax_corss_entrophy,
```
Comment (anj-s) [inline official/resnet/keras/keras_imagenet_main.py]: typo: softmax_cross_entropy
maybe say _logits to specify that this is a logits loss?

File: official/resnet/keras/keras_imagenet_main.py [hunk 2]
```diff
@@ -187,6 +186,16 @@ def run_imagenet_with_keras(flags_obj):
                                      steps_per_epoch, samples_per_second))
 
 
+def softmax_corss_entrophy(y_true, y_pred):
+  """A cost function replicating tf's sparse_softmax_cross_entropy
```
Comment (anj-s) [inline official/resnet/keras/keras_imagenet_main.py]: maybe say "loss function" instead of "cost"? 
Also I would not say replicating since we are calling TensorFlow's loss function.

File: official/resnet/keras/keras_imagenet_main.py (hunk 1 above)
Comment (anj-s) [inline official/resnet/keras/keras_imagenet_main.py]: Let us also add a comment about the fact that we are using logits loss.

File: official/resnet/keras/keras_imagenet_main.py [hunk 3]
```diff
@@ -84,7 +84,6 @@ def parse_record_keras(raw_record, is_training, dtype):
       is_training=is_training)
 
   image = tf.cast(image, dtype)
-  label = tf.sparse_to_dense(label, (imagenet_main._NUM_CLASSES,), 1)
```
Comment (anj-s) [inline official/resnet/keras/keras_imagenet_main.py]: Don't we need to remove this for the synthetic_input_fn as well?

**Comment by anj-s** [comment]
Once we unify the loss functions we can remove the Keras specific parse function.

File: official/resnet/keras/keras_imagenet_main.py (hunk 1 above)
Comment (seemuch) [reply official/resnet/keras/keras_imagenet_main.py]: Done

File: official/resnet/keras/keras_imagenet_main.py (hunk 2 above)
Comment (seemuch) [reply official/resnet/keras/keras_imagenet_main.py]: Done

File: official/resnet/keras/keras_imagenet_main.py (hunk 1 above)
Comment (seemuch) [reply official/resnet/keras/keras_imagenet_main.py]: I already changed the function name. Do you think the comment is still needed?

File: official/resnet/keras/keras_imagenet_main.py (hunk 3 above)
Comment (seemuch) [reply official/resnet/keras/keras_imagenet_main.py]: Done.

File: official/resnet/keras/keras_imagenet_main.py [hunk 4]
```diff
@@ -187,6 +185,16 @@ def run_imagenet_with_keras(flags_obj):
                                      steps_per_epoch, samples_per_second))
 
 
+def softmax_corssentropy_with_logits(y_true, y_pred):
```
Comment (anj-s) [inline official/resnet/keras/keras_imagenet_main.py]: s/corss/cross

Comment (seemuch) [reply official/resnet/keras/keras_imagenet_main.py]: Done
